sklearn/mixture/tests/test_gmm.py

```diff
@@ -327,6 +328,32 @@ def test_1d_1component():
         assert_array_almost_equal(g.bic(X), g_full_bic)
 
 
+def assert_fit_predict_correct(model, X):
+
+    predictions_1 = model.fit(X).predict(X)
+    predictions_2 = model.fit_predict(X)
+
+    assert adjusted_rand_score(predictions_1, predictions_2) == 1.0
```

Comment:
That makes it explicit that the absolute values of the labels are just cluster membership symbols and that we only want to test the equality of the partitioning, not the labeling itself. This check should be robust to independent random model inits even if we don't try to change the rng seed here.